modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/BrokerAuthConfiguration.java

```diff
@@ -292,4 +302,168 @@ public void setSize(int size) {
+
+        private String hostName = "localhost";
+
+        private String baseDN = "dc=example,dc=com";
+
+        private String usernameSearchFilter = "(uid=?)";
+
+        private LdapPlainConfiguration plain = new LdapPlainConfiguration();
+
+        private LdapSslConfiguration ssl = new LdapSslConfiguration();
+
+        public String getHostName() {
+            return hostName;
+        }
+
+        public void setHostName(String hostName) {
+            this.hostName = hostName;
+        }
+
+        public String getBaseDN() {
+            return baseDN;
+        }
+
+        public void setBaseDN(String baseDN) {
+            this.baseDN = baseDN;
+        }
+
+        public String getUsernameSearchFilter() {
+            return usernameSearchFilter;
+        }
+
+        public void setUsernameSearchFilter(String usernameSearchFilter) {
+            this.usernameSearchFilter = usernameSearchFilter;
+        }
+
+        public LdapPlainConfiguration getPlain() {
+            return plain;
+        }
+
+        public void setPlain(LdapPlainConfiguration plain) {
+            this.plain = plain;
+        }
+
+        public LdapSslConfiguration getSsl() {
+            return ssl;
+        }
+
+        public void setSsl(LdapSslConfiguration ssl) {
+            this.ssl = ssl;
+        }
+    }
+
+    /**
+     * Represents Ldap configuration for plain connections (ssl disabled).
+     */
+    public static class LdapPlainConfiguration {
+
+        private int port = 10389;
+
+        public int getPort() {
+            return port;
+        }
+
+        public void setPort(int port) {
+            this.port = port;
+        }
+    }
+
+    /**
+     * Represents Ldap ssl configuration.
+     */
+    public static class LdapSslConfiguration {
+
+        private boolean enabled = false;
+
+        private int port = 10636;
+
+        private String protocol = "TLS";
+
+        private LdapSslTrustStoreConfiguration trustStore = new LdapSslTrustStoreConfiguration();
+
+        public boolean isEnabled() {
+            return enabled;
+        }
+
+        public void setEnabled(boolean enabled) {
+            this.enabled = enabled;
+        }
+
+        public int getPort() {
+            return port;
+        }
+
+        public void setPort(int port) {
+            this.port = port;
+        }
+
+        public String getProtocol() {
+            return protocol;
+        }
+
+        public void setProtocol(String protocol) {
+            this.protocol = protocol;
+        }
+
+        public LdapSslTrustStoreConfiguration getTrustStore() {
+            return trustStore;
+        }
+
+        public void setTrustStore(LdapSslTrustStoreConfiguration trustStore) {
+            this.trustStore = trustStore;
+        }
+    }
+
+    /**
+     * Represents ssl trust store configuration for Ldap.
+     */
+    public static class LdapSslTrustStoreConfiguration {
+
+        private String type = "JKS";
+
```

Comment:
unwanted spaces between

Reply:
Those spaces were added to be consistent with the other existing classes in the same java file

Reply:
1 new line is after the variable definition is fine IMO.